pkg/tsdb/opentsdb/opentsdb.go

```diff
@@ -3,53 +3,122 @@ package opentsdb
 import (
 	"context"
 	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
 	"path"
 	"strconv"
 	"strings"
 	"time"
 
-	"golang.org/x/net/context/ctxhttp"
-
 	"encoding/json"
-	"io/ioutil"
-	"net/http"
-	"net/url"
 
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/datasource"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
 	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana/pkg/components/simplejson"
 	"github.com/grafana/grafana/pkg/infra/httpclient"
 	"github.com/grafana/grafana/pkg/infra/log"
-	"github.com/grafana/grafana/pkg/models"
-	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin/coreplugin"
+	"github.com/grafana/grafana/pkg/registry"
 	"github.com/grafana/grafana/pkg/setting"
+	"golang.org/x/net/context/ctxhttp"
 )
 
 type OpenTsdbExecutor struct {
 	httpClientProvider httpclient.Provider
+
+	im  instancemgmt.InstanceManager
+	cfg *setting.Cfg
+}
+
+type OpenTsdbService struct {
```

Comment:
Since package is named `opentsdb` would suggest naming this `Service`. WDYT?

There's an additional struct `OpenTsdbExecutor` where the idea of "executor" feels somewhat related to using non-SDK contracts/old way. Since you already have a struct for `Service` you could basically move everything from `OpenTsdbExecutor` into `Service`. An alternative if you would like to distinguish between service and plugin struct might want to rename `OpenTsdbExecutor` to `Plugin`. WDYT?

Reply:
I think the executor should go, I moved the rest to Service